### PR TITLE
[Spring] fix Paginated without params (fix #15265)

### LINKED ISSUE #15265: [BUG][Spring] x-spring-paginated fails if no params is defined in the operation
#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
   - swagger.txt (openapi 2.0.0) in oas3 we hava the same problem
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
   - [ ] yes
- [x] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists? 
   - [ ] All 6.x versions up to 6.5.0 included
- [x] Have you searched for related issues/PRs? 
   - [ ] yes
- [x] What's the actual output vs expected output?
   - In class PrestacionesApiDelegate interface methods should be : ``obtenerDatosEncuestaDecesos(final Pageable pageable);`` not ``obtenerDatosEncuestaDecesos(, final Pageable pageable); ``
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

##### Description

When using generator spring/spring-cloud and delegatePattern is active:
```
                <generatorName>spring</generatorName>
                <library>spring-cloud</library>
                <configOptions>
                  <delegatePattern>true</delegatePattern>
                </configOptions>
```
And try to compile an oas2/oas3 spec with x-spring-paginated: true and no params, generator fails due to the template is introducing a "," prior to Pageable object
``obtenerDatosEncuestaDecesos(, final Pageable pageable); ``

The problem seems to came from this point at [apiDelegate.mustache](https://github.com/OpenAPITools/openapi-generator/blob/v6.5.0/modules/openapi-generator/src/main/resources/JavaSpring/apiDelegate.mustache#L67) where the "," is introduced always
##### openapi-generator version

All 6.x versions up to 6.5.0 included

##### OpenAPI declaration file content or url
[swagger.txt](https://github.com/OpenAPITools/openapi-generator/files/11272676/swagger.txt)

##### Generation Details

```
                <generatorName>spring</generatorName>
                <library>spring-cloud</library>
                <configOptions>
                  <delegatePattern>true</delegatePattern>
                </configOptions>
```

##### Related issues/PRs

N/A

##### Suggest a fix

check if no params is defined, and remove "," in this cases.
Problem came from this line in [apiDelegate.mustache](https://github.com/OpenAPITools/openapi-generator/blob/v6.5.0/modules/openapi-generator/src/main/resources/JavaSpring/apiDelegate.mustache#L67) where the "," is introduced always

### PR DESCRIPTION
**Pull request by borsch**
fix for https://github.com/OpenAPITools/openapi-generator/issues/15265

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] In case you are adding a new generator, run the following additional script : 
  ```
  ./bin/utils/ensure-up-to-date.sh
  ``` 
  Commit all changed files.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.3.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

@cachescrubber (2022/02) @welshm (2022/02) @MelleD (2022/02) @atextor (2022/02) @manedev79 (2022/02) @javisst (2022/02) @borsch (2022/02) @banlevente (2022/02) @Zomzog (2022/09)

### CODE REVIEW

**Review by welshm — COMMENTED** [review]
Did none of the samples change from this?

File: modules/openapi-generator/src/main/resources/JavaSpring/api.mustache [hunk 1]
```diff
@@ -244,20 +244,20 @@ public interface {{classname}} {
...
...lt-interface}}default {{/jdk8-default-interface}} {{#responseWrapper}}{{.}}<{{/responseWrapper}}{{#useResponseEntity}}ResponseEntity<{{/useResponseEntity}}{{>returnTypes}}{{#useResponseEntity}}>{{/useResponseEntity}}{{#responseWrapper}}>{{/responseWrapper}} {{operationId}}({{#allParams}}{{^isFile}}{{^isBodyParam}}{{>optionalDataType}}{{/isBodyParam}}{{#isBodyParam}}{{^reactive}}{{{dataType}}}{{/reactive}}{{#reactive}}{{^isArray}}Mono<{{{dataType}}}>{{/isArray}}{{#isArray}}Flux<{{{baseType}}}>{{/isArray}}{{/reactive}}{{/isBodyParam}}{{/isFile}}{{#isFile}}{{#reactive}}Flux<Part>{{/reactive}}{{^reactive}}MultipartFile{{/reactive}}{{/isFile}} {{paramName}}{{^-last}}, {{/-last}}{{/allParams}}{{#reactive}}{{#hasParams}}, {{/hasParams}}{{#springFoxDocumentationProvider}}@ApiIgnore{{/springFoxDocumentationProvider}} final ServerWebExchange exchange{{/reactive}}{{#vendorExtensions.x-spring-paginated}}{{#hasParams}},{{/hasParams}}{{^hasParams}}{{#reactive}},{{/reactive}}{{/hasParams}} {{#springFoxDocumentationProvider}}@ApiIgnore{{/springFoxDocumentationProvider}} final Pageable pageable{{/vendorExtensions.x-spring-paginated}}){{#unhandledException}} throws Exception{{/unhandledException}} {
```
[the remark this answers is not in the document]
Comment (borsch) [reply modules/openapi-generator/src/main/resources/JavaSpring/api.mustache:254]: image

**Comment by borsch** [comment]
> Did none of the samples change from this?

@welshm yeah, because our common spec files that are used for generation doesn't have operations with pagination & **without** parameters

**Comment by welshm** [comment]
> > Did none of the samples change from this?
> 
> @welshm yeah, because our common spec files that are used for generation doesn't have operations with pagination & **without** parameters

Should we add a sample? Code change looks good to me otherwise 🚀

**Review by welshm — APPROVED** [review]

**Comment by borsch** [comment]
> > > Did none of the samples change from this?
> > 
> > 
> > @welshm yeah, because our common spec files that are used for generation doesn't have operations with pagination & **without** parameters
> 
> Should we add a sample? Code change looks good to me otherwise 🚀

I've updated samples with one endpoint

**Pull request merged**